src/main/java/ar/com/carloscurotto/storm/complex/topology/propagator/gloss/MessageSender.java

```diff
@@ -0,0 +1,121 @@
+package ar.com.carloscurotto.storm.complex.topology.propagator.gloss;
+
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.jms.JMSException;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.commons.lang3.Validate;
+
+import ar.com.carloscurotto.storm.complex.service.OpenAwareService;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.destination.IDestination;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.exception.GlossJmsException;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.exception.GlossMarshalException;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.ExcpTradeStatusMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.NormalTradeStatusMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.TradeCommentsMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.TradeMessage;
+
+
+/**
+ * Sends the message into the queue. Knows what queue to use and knows how to marshall the messages.
+ *
+ * @author D540601
+ */
+public class MessageSender extends OpenAwareService<Messages, Void> {
+
+    protected Map<Class<? extends TradeMessage>, Marshaller> marshallers = new HashMap<Class<? extends TradeMessage>, Marshaller>();
+
+    /**
+     * Destination where the messages are directed.
+     */
+    protected IDestination destination;
+    
+    @Override
+    protected void doOpen() {
+        try {
+            marshallers.put(NormalTradeStatusMessage.class,
+                    JAXBContext.newInstance(NormalTradeStatusMessage.class).createMarshaller());
+            marshallers.put(ExcpTradeStatusMessage.class,
+                    JAXBContext.newInstance(ExcpTradeStatusMessage.class).createMarshaller());
+            marshallers.put(TradeCommentsMessage.class,
+                    JAXBContext.newInstance(TradeCommentsMessage.class).createMarshaller());
+        } catch (JAXBException e) {
+            throw new RuntimeException("MessageSender can't create the marshallers", e);
+        }
+
+    }
+
+    @Override
+    protected void doClose() {
+        try {
+            destination.close();
+        } catch (JMSException e) {
+            throw new RuntimeException("MessageSender can't close destination", e);
+        }
+    }
+
+    @Override
+    protected Void doExecute(Messages theMessages) {
```

Comment:
me gustaria que sea:
protected void doExecute(List<Message> theMessages) {
    for(Message message : theMessages) send(message.getText());
}